File: yixiang_be/src/main/java/com/tongji/circle/model/Circle.java
package com.tongji.circle.model;

import lombok.Data;
import java.time.Instant;

@Data
public class Circle {
    private Long id;
    private String name;
    private String description;
    private String avatarUrl;
    private Long ownerId;
    private String visibility;
    private String status;
    private String category;
    private int memberCount;
    private int postCount;
    private Instant createdAt;
}

File: yixiang_fe/src/types/circle.ts
export interface CircleSummary {
  id: number;
  name: string;
  description: string | null;
  avatarUrl: string | null;
  category: string | null;
  visibility: 'PUBLIC' | 'PRIVATE';
  memberCount: number;
  postCount: number;
  joined: boolean;
}

export interface MemberSummary {
  userId: number;
  nickname: string;
  avatar: string | null;
  role: 'OWNER' | 'ADMIN' | 'MEMBER';
}

export interface CircleDetail extends CircleSummary {
  createdAt: string;
  myRole: 'OWNER' | 'ADMIN' | 'MEMBER' | null;
  topMembers: MemberSummary[];
}

export interface CircleListResponse {
  items: CircleSummary[];
  total: number;
  page: number;
  size: number;
}
